packages/libsql/src/lib/LibSqlClient.ts

```diff
@@ -0,0 +1,203 @@
+import { randomUUID } from "crypto"
+import type { Client, Transaction } from "@libsql/client"
+import { IEventData, ITimelineEvent } from "@equinox-js/core"
+
+type LibSqlWriteResult =
+  | { type: "Written"; position: bigint; snapshot_etag?: string }
+  | { type: "ConflictUnknown" }
+export type Format = string
+
+type Trx = {
+  execute: Client['execute']
+}
+
+export class LibSqlWriter {
+  constructor(private readonly client: Client) {}
+
+  /**
+   * Reads the current version of the stream
+   * version is 0 based, an empty stream has version 0
+   */
+  async readStreamVersion(streamName: string, trx: Trx = this.client): Promise<bigint> {
+    const result = await trx.execute({
+      sql: "select max(position) from messages where stream_name = ?",
```

Comment:
would index be a better name now? aligns with i around the rest of the world, or does position have some specific meaning?

Reply:
`arr[0]` = the first event... If it's index then the first event has index = 0 and the stream is at version = 1

Better to have something that conveys the first event is 1?

Reply:
nvm, just changed it to be an actual index. Though some might argue index is 1 based and 0 based is offsets